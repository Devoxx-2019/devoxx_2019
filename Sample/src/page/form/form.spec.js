import React from "react";
import { render } from "react-testing-library";
import Component from "./form";

describe("Form test suite", () => {
  const users = [
    {
      login: "login1",
      avatar_url: "http://avatar1.jpg",
      html_url: "http://comptegithub.com"
    },
    {
      login: "login2",
      avatar_url: "http://avatar2.jpg",
      html_url: "http://comptegithub.com"
    },
    {
      login: "login3",
      avatar_url: "http://avatar3.jpg",
      html_url: "http://comptegithub.com"
    }
  ];

  const inputs = {
    githubAccount: {
      value: "test",
      id: "githubAccount",
      message: "erreur de saisie"
    }
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <Component
        users={users}
        inputs={inputs}
        onChange={() => {}}
        onClick={() => {}}
        hasSubmitOnce={true}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
