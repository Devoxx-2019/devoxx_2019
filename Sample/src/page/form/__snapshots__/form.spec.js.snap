// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form test suite renders correctly 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <div
      class="form__line"
    >
      <span
        class="form__label"
      />
      <div
        class="form__input-container"
      >
        <input
          id="githubAccount"
          name="githubAccount"
          value="test"
        />
        <span
          class="form__error-message"
        >
          erreur de saisie
        </span>
      </div>
    </div>
    <div
      class="form__line form__line--alone"
    >
      <button
        class="btn btn--search"
        data-testid="button"
      >
        Rechercher
      </button>
    </div>
  </form>
  <div
    class="user-list"
    data-testid="userlist"
  >
    <div
      class="user-card"
      data-testid="usercard"
    >
      <a
        class="user-card__link"
        href="http://comptegithub.com"
      >
        <img
          alt="login1"
          class="user-card__avatar"
          src="http://avatar1.jpg"
        />
        <span
          class="user-card__login"
        >
          login1
        </span>
      </a>
    </div>
    <div
      class="user-card"
      data-testid="usercard"
    >
      <a
        class="user-card__link"
        href="http://comptegithub.com"
      >
        <img
          alt="login2"
          class="user-card__avatar"
          src="http://avatar2.jpg"
        />
        <span
          class="user-card__login"
        >
          login2
        </span>
      </a>
    </div>
    <div
      class="user-card"
      data-testid="usercard"
    >
      <a
        class="user-card__link"
        href="http://comptegithub.com"
      >
        <img
          alt="login3"
          class="user-card__avatar"
          src="http://avatar3.jpg"
        />
        <span
          class="user-card__login"
        >
          login3
        </span>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
